packages/nextra/src/loader.ts

```diff
@@ -269,102 +248,27 @@ export default MDXContent`
       // Remove the only `index` route
       .replace(/^index$/, '')
 
-  const convertToRelativePath = (filePath: string): string => {
-    const relative = path.relative(path.dirname(mdxPath), filePath)
-    const finalPath = relative.startsWith('.') ? relative : `./${relative}`
-    return slash(finalPath)
-  }
-
-  const dynamicMetaResolver = dynamicMetaItems.length
-    ? `if (typeof window === 'undefined') {
-  globalThis.__nextra_resolvePageMap__ = async () => {
-    const clonedPageMap = JSON.parse(JSON.stringify(__nextra_pageOpts__.pageMap))
-  
-    const executePromises = []
-    const importPromises = [
-      ${dynamicMetaItems
-        .map(
-          ({
-            metaFilePath,
-            metaObjectKeyPath,
-            metaParentKeyPath
-          }) => `import('${convertToRelativePath(metaFilePath)}').then(m => {
-        const getMeta = Promise.resolve(m.default()).then(metaData => {
-          const meta = clonedPageMap${metaObjectKeyPath}
-          meta.data = metaData
```

Comment:
This change causes a bunch of errors like the following:

```
swr-site:dev: warn  - ../../node_modules/.pnpm/file+packages+nextra_q76c2b4vyoegvsbrcwkfvimnai/node_modules/nextra/dist/setup-page.mjs
swr-site:dev: Critical dependency: the request of a dependency is an expression
```

Because previously all the imports are static and inlined as code, but now they're dynamic so webpack can't figure out the module graph without running the code.

Reply:
I think I'll try to refactor it to make it both type safe and static. Although it's tricky due to the nature of webpack loaders.